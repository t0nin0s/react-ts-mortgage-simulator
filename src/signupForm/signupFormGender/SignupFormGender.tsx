import * as React from 'react';
import * as FontAwesome from 'react-fontawesome';
import Card from '../../components/Card/Card';
import MyButton from '../../components/Button/Button';
const styles = require('../signupForm.css');

interface SignupFormGenderProps {
  gender: string;
  handleOnChange: (event: React.MouseEvent<HTMLInputElement>) => void;
  handleNext: () => void;
}

class SignupFormGender extends React.Component<SignupFormGenderProps, { error: boolean } > {
  constructor() {
    super();
    this.state = {
      error: false
    };
  }

  validate = () => {
    if (this.props.gender) {
      this.setState({
        error: false
      });
      this.props.handleNext();
    } else {
      this.setState({
        error: true
      });
    }
  }

  render() {
    return (
      <Card
        formTitle="Sign up"
        icon="question-circle-o"
        footer={true}
      >
        <div className={styles.form__question}>
          Please tell us your gender...
        </div>
        <div className={styles.form__body}>
          <MyButton
            id="gender"
            value="male"
            text="Male"
            icon="mars"
            large={true}
            handleOnChange={this.props.handleOnChange}
          />
          <MyButton
            id="gender"
            value="female"
            text="Female"
            icon="venus"
            large={true}
            handleOnChange={this.props.handleOnChange}
          />
        </div>
        { this.state.error &&
          <div className={styles.error_message}>
            Please select first your gender...
          </div>
        }
        <div className={styles.form__submit}>
          <FontAwesome
            name="arrow-circle-right"
            size="3x"
            style={{ color: '#ECECEC' }}
            onClick={this.validate}
          />
        </div>
      </Card>
    );
  }
}

export default SignupFormGender;
