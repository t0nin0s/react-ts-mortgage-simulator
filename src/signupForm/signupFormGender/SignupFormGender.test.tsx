import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SignupFormGender from './SignupFormGender'

describe('SignupFormGender renders as expected', () => {
  it('should render as expected', () => {
    const component = shallow(
      <SignupFormGender
        gender=''
        handleOnChange={jest.fn()}
        handleNext={jest.fn()} />
    )
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
