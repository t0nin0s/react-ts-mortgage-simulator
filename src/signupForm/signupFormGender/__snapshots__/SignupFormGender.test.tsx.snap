// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupFormGender renders as expected should render as expected 1`] = `
<Card
  footer={true}
  formTitle="Sign up"
  icon="question-circle-o"
>
  <div>
    Please tell us your gender...
  </div>
  <div>
    <MyButton
      handleOnChange={[Function]}
      icon="mars"
      id="gender"
      large={true}
      text="Male"
      value="male"
    />
    <MyButton
      handleOnChange={[Function]}
      icon="venus"
      id="gender"
      large={true}
      text="Female"
      value="female"
    />
  </div>
  <div>
    <FontAwesome
      name="arrow-circle-right"
      onClick={[Function]}
      size="3x"
      style={
        Object {
          "color": "#ECECEC",
        }
      }
    />
  </div>
</Card>
`;
